client/client-multi/client-c/src/kaa/platform-impl/common/sha.c

```diff
@@ -24,7 +24,7 @@
 
 kaa_error_t ext_calculate_sha_hash(const char *data, size_t data_size, kaa_digest digest)
 {
-    if (!digest || !data || !data_size) {
+    if (!digest || (!data && data_size)) {
```

Comment:
And also, don't forget to change function documentation in the header file. All changes ***must*** be reflected there.